### DISCUSSION TITLE
Update all existing abs metadata file for write Tag information?

### DISCUSSION THREAD

**Discussion by Hallo951** [advplyr/audiobookshelf#1510] [Q&A]
How do I update all abs metadata (approx. 4000 pieces) of my audiobook library so that the tag information is written in all abs metadata?

**Comment by hobesman** [comment]
Just to clarify, are you wanting to take the metadata abs has, and embed it into your files? Or are you wanting to take the tags already in the files and make abs use that info? Or what source are you wanting to use, and where are you wanting to write that data?

**Reply by Hallo951** [reply]
With the latest update of Audiobookshelf, not only the categories but also the tags of the individual audiobooks are now written into the abs metadata files when, for example, a new audiobook is added to the database and scrapped using any provider (e.g. Audible). This was not the case before the update. I think the tags were stored in the database itself but not in the abs metadata files. I think this is great and would now like to update all the metadata (in my case it is the abs files) of all my audio books so that not only the categories but also the tags are stored in the abs files. That is all. So the question was, how do I update all my abs metadata files?

**Reply by Hallo951** [reply]
I just had an idea. One could simply add (append) a new tag to all audio books, for example with the name "Temp", and thus force the update of the abs metadata files. In this process, all abs files would be rewritten and thus all other tags from the database would be transferred to the abs metadata files.

The only problem I have is how do I delete the tag "Temp" from all abs metadata files after this process? I can attach a new tag to existing metadata files, but how do I delete a specific tag from all metadata files?

**Reply by advplyr** [reply]
You could batch add the tag and save. That will update the metadata files.
Then in the metadata utils in config you can delete the tag
![image](https://user-images.githubusercontent.com/67830747/219478440-2cdfeb4b-70ee-4e3c-ad17-b208dc30427b.png)

**Reply by Hallo951** [reply]
:-) thx
